workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -186,17 +211,63 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
       eventActionsChannel = eventActionsChannel
     )
     val rendering = interceptor.intercept(workflow, this)
-      .render(props, state, RenderContext(context, workflow))
+      .render(props, state.value, RenderContext(context, workflow))
     context.freeze()
 
+    commitAndUpdateScopes()
+
+    return rendering
+  }
+
+  @Composable
+  private fun RenderingWithStateType(
+    workflow: StatefulWorkflow<PropsT, StateT, OutputT, RenderingT>,
+    props: PropsT,
+    setRendering: (RenderingT) -> Unit
+  ): Unit {
+    key(props) {
```

Comment:
If you're comparing props anyway, I don't see the reason for putting this in a key function (and if it is necessary, it should probably be an implementation detail of UpdatePropsAndState).

Reply:
removed.